community/detector/cves/CVE-2021-29441/src/main/java/com/google/tsunami/plugins/detectors/cves/cve202129441/CVE202129441VulnDetector.java

```diff
@@ -0,0 +1,145 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.tsunami.plugins.detectors.cves.cve202129441;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.google.common.net.HttpHeaders.*;
+import static com.google.tsunami.common.net.http.HttpRequest.get;
+import static com.google.tsunami.common.net.http.HttpRequest.post;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.flogger.GoogleLogger;
+import com.google.protobuf.util.Timestamps;
+import com.google.tsunami.common.data.NetworkServiceUtils;
+import com.google.tsunami.common.net.http.HttpClient;
+import com.google.tsunami.common.net.http.HttpHeaders;
+import com.google.tsunami.common.net.http.HttpResponse;
+import com.google.tsunami.common.time.UtcClock;
+import com.google.tsunami.plugin.PluginType;
+import com.google.tsunami.plugin.VulnDetector;
+import com.google.tsunami.plugin.annotations.PluginInfo;
+import com.google.tsunami.proto.DetectionReport;
+import com.google.tsunami.proto.DetectionReportList;
+import com.google.tsunami.proto.DetectionStatus;
+import com.google.tsunami.proto.NetworkService;
+import com.google.tsunami.proto.Severity;
+import com.google.tsunami.proto.TargetInfo;
+import com.google.tsunami.proto.Vulnerability;
+import com.google.tsunami.proto.VulnerabilityId;
+
+import java.io.IOException;
+import java.time.Clock;
+import java.time.Instant;
+import javax.inject.Inject;
+
+@PluginInfo(
+		type = PluginType.VULN_DETECTION,
+		name = "CVE-2021-29441",
+		version = "0.1",
+		description = "Nacos is a platform designed for dynamic service discovery and configuration and service " +
+				"management. In Nacos before version 1.4.1, when configured to use authentication (-Dnacos.core.auth" +
+				".enabled=true) Nacos uses the AuthFilter servlet filter to enforce authentication. This filter has a" +
+				" backdoor that enables Nacos servers to bypass this filter and therefore skip authentication checks." +
+				" This mechanism relies on the user-agent HTTP header so it can be easily spoofed. This issue may allow" +
+				" any user to carry out any administrative tasks on the Nacos server.",
+		author = "hh-hunter",
+		bootstrapModule = CVE202129441DetectorBootstrapModule.class)
+
+public final class CVE202129441VulnDetector implements VulnDetector {
+
+	private static final GoogleLogger logger = GoogleLogger.forEnclosingClass();
+
+	private static final String CHECK_VUL_PATH = "nacos/v1/auth/users/?pageNo=1&pageSize=9";
+
+	public static final String DETECTION_STRING = "pageItems";
+
+	private final HttpClient httpClient;
+
+	private final Clock utcClock;
```

Comment:
Code style: Please put `static final` members in a single block and other members in a separate block.